template/flake.nix

```diff
@@ -2,80 +2,92 @@
   description = "Soxin template flake";
 
   inputs = {
-    nixos.url = "nixpkgs/nixos-20.09";
-    master.url = "nixpkgs/master";
+    deploy-rs.url = github:serokell/deploy-rs;
+    nixpkgs.url = github:NixOS/nixpkgs/release-21.05;
+    nur.url = github:nix-community/NUR;
+    sops-nix.url = github:Mic92/sops-nix;
+    unstable.url = github:NixOS/nixpkgs/nixos-unstable;
+    utils.url = github:gytis-ivaskevicius/flake-utils-plus/v1.1.0;
+
     home-manager = {
-      url = "github:nix-community/home-manager/release-20.09";
-      inputs.nixpkgs.follows = "nixos";
+      url = "github:nix-community/home-manager";
+      inputs.nixpkgs.follows = "nixpkgs";
     };
+
     soxin = {
-      url = "github:SoxinOS/soxin";
+      url = path:../.;
       inputs = {
-        nixpkgs.follows = "nixos";
+        deploy-rs.follows = "deploy-rs";
         home-manager.follows = "home-manager";
+        nixpkgs.follows = "nixpkgs";
+        nur.follows = "nur";
+        sops-nix.follows = "sops-nix";
+        unstable.follows = "unstable";
+        utils.follows = "utils";
       };
     };
-    futils.url = "github:numtide/flake-utils";
   };
 
-  outputs = { self, nixos, master, home-manager, soxin, futils } @ inputs:
+  outputs = inputs@{ self, soxin, nixpkgs, ... }:
     let
-      inherit (nixos) lib;
-      inherit (nixos.lib) recursiveUpdate;
-      inherit (futils.lib) eachDefaultSystem;
-
-      pkgImport = pkgs: system:
-        import pkgs {
-          inherit system;
-          overlays = lib.attrValues self.overlays;
-          config = { allowUnfree = true; };
-        };
+      # Enable deploy-rs support
+      withDeploy = true;
 
-      pkgset = system: {
-        nixos = pkgImport nixos system;
-        master = pkgImport master system;
-      };
+      # Enable sops support
+      withSops = true;
+
+      inherit (nixpkgs) lib;
+      inherit (lib) optionalAttrs recursiveUpdate singleton;
 
-      multiSystemOutputs = eachDefaultSystem (system:
-        let
-          pkgset' = pkgset system;
-          osPkgs = pkgset'.nixos;
-          pkgs = pkgset'.master;
-        in
-        {
-          devShell = pkgs.mkShell {
-            buildInputs = with pkgs; [
-              git
-              nixpkgs-fmt
-              pre-commit
-            ];
-          };
-
-          packages = soxin.lib.overlaysToPkgs self.overlays pkgs;
-        }
-      );
-
-      outputs = {
-        overlay = self.overlays.packages;
-
-        overlays = import ./overlays;
-
-        nixosModules = recursiveUpdate (import ./modules) {
-          profiles = import ./profiles;
+      # Channel definitions. `channels.<name>.{input,overlaysBuilder,config,patches}`
+      channels = {
```

Comment:
I __really__ don't like the naming conventions here. Maybe we should open a discussion upstream to see if we can get it changed.

Reply:
The channels? I think they make total sense especially when moving away from nix to flakes.

Reply:
Yes, they do make sense, but it might confuse someone moving from channels to flakes even more.

Reply:
I'm not sure that's the case tbh. I doubt anyone was using the channels in the first place, but for those that did will also be well aware of what it means.